CMakeLists.txt

```diff
@@ -367,7 +405,7 @@ if (NOT ENSMALLEN_FOUND)
     # Release versions will have ensmallen packaged with the release so we can
     # just reference that.
     if (EXISTS "${CMAKE_SOURCE_DIR}/src/mlpack/core/optimizers/ensmallen/ensmallen.hpp")
-      set(MLPACK_INCLUDE_DIRS ${MLPACK_INCLUDE_DIRS} ${ARMADILLO_INCLUDE_DIRS}
+      set(MLPACK_INCLUDE_DIRS ${MLPACK_INCLUDE_DIRS} ${ARMADILLO_INCLUDE_DIR}
```

Comment:
We do need to use `ARMADILLO_INCLUDE_DIRS` here, since that could actually refer to several directories needed to properly capture Armadillo's dependencies (like HDF5).  Maybe reworking the `FindArmadillo.cmake` script is the better option here?